staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -50,6 +50,11 @@ type RequestScope struct {
 	Serializer runtime.NegotiatedSerializer
 	runtime.ParameterCodec
 
+	// StandardSerializers, if set, restricts which serializers can be used when
+	// we aren't transforming the output (into Table or PartialObjectMetadata).
+	// Used only by CRDs which do not yet support Protobuf.
```

Comment:
I guess I find the changes here a touch surprising, couldn't CRDs have produced a wrapper for the negotiated serializer? cc @deads2k

Reply:
No, we actually need to vary based on request headers and protocols above this code (in the rest stack).  Negotiated serializer doesn’t actually perform HTTP negotiation and we need to convey “CRD doesn’t allow proto on X, but does when X becomes Y”.